internal/testhelpers/e2e/shell.go

```diff
@@ -0,0 +1,31 @@
+package e2e
+
+import (
+	"strings"
+
+	"github.com/ActiveState/cli/internal/osutils"
+)
+
+type Shell string
+
+const (
+	Bash Shell = "bash"
+	Zsh        = "zsh"
+	Tcsh       = "tcsh"
+	Fish       = "fish"
+	Cmd        = "cmd.exe"
+)
+
+// QuoteCommand constructs and returns a command line string from the given list of arguments.
+// The returned string can be passed to the given shell for evaluation.
+func QuoteCommand(shell Shell, args ...string) string {
```

Comment:
Would this work the same on cmd.exe?

Reply:
The behavior is not exactly the same with cmd.exe vs bash, but the idea is the same. As you saw in the the function body, cmd.exe has its own quoter.